From eccc504809b665ec6bb37d798f9627186b6d2cff Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Tue, 4 Oct 2022 10:47:05 +0800
Subject: [PATCH 016/115] mtk: mac80211: Set TWT Information Frame Disabled bit
 as 1.

This modification means that current implementation do not support twt information frame.
---
 net/mac80211/s1g.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/net/mac80211/s1g.c b/net/mac80211/s1g.c
index 1f68df6e..f61a7957 100644
--- a/net/mac80211/s1g.c
+++ b/net/mac80211/s1g.c
@@ -102,6 +102,7 @@ ieee80211_s1g_rx_twt_setup(struct ieee80211_sub_if_data *sdata,
 	struct ieee80211_twt_params *twt_agrt = (void *)twt->params;
 
 	twt_agrt->req_type &= cpu_to_le16(~IEEE80211_TWT_REQTYPE_REQUEST);
+	twt->control |= IEEE80211_TWT_CONTROL_RX_DISABLED;
 
 	/* broadcast TWT not supported yet */
 	if (twt->control & IEEE80211_TWT_CONTROL_NEG_TYPE_BROADCAST) {
-- 
2.45.2

